cline_utils/dependency_system/utils/tracker_utils.py

```
@@ -0,0 +1,189 @@
+    Read a tracker file and parse its contents. Caches based on path and mtime.
+    Args:
+        tracker_path: Path to the tracker file
+    Returns:
+        Dictionary with keys, grid, and metadata, or empty structure on failure.
+    """
+    tracker_path = normalize_path(tracker_path)
+    if not os.path.exists(tracker_path):
+        logger.debug(f"Tracker file not found: {tracker_path}. Returning empty structure.")
+        return {"keys": {}, "grid": {}, "last_key_edit": "", "last_grid_edit": ""}
+    try:
+        with open(tracker_path, 'r', encoding='utf-8') as f: content = f.read()
```

Ruff (0.8.2)

Multiple statements on one line (colon)

(E701)